edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesExperimentalPodParametersTest.cs

```diff
@@ -168,5 +169,92 @@ public void ParsesSomeResourcesExperimentalOptions()
             parameters.Resources.ForEach(resources => Assert.Equal(new Dictionary<string, ResourceQuantity> { ["memory"] = new ResourceQuantity("128Mi") }, resources.Limits));
             parameters.Resources.ForEach(resources => Assert.Equal(new Dictionary<string, ResourceQuantity> { ["cpu"] = new ResourceQuantity("250m") }, resources.Requests));
         }
+
+        [Fact]
+        public void ParsesNoneVolumesExperimentalOptions()
+        {
+            var experimental = new Dictionary<string, JToken>
+            {
+                ["k8s-experimental"] = JToken.Parse("{ volumes: null }")
+            };
+
+            var parameters = KubernetesExperimentalCreatePodParameters.Parse(experimental).OrDefault();
+
+            Assert.False(parameters.Volumes.HasValue);
+        }
+
+        [Fact]
+        public void ParsesEmptyVolumesExperimentalOptions()
+        {
+            var experimental = new Dictionary<string, JToken>
+            {
+                ["k8s-experimental"] = JToken.Parse("{ volumes: [  ] }")
+            };
+
+            var parameters = KubernetesExperimentalCreatePodParameters.Parse(experimental).OrDefault();
+
+            Assert.True(parameters.Volumes.HasValue);
+            parameters.Volumes.ForEach(Assert.Empty);
+        }
+
+        [Fact]
+        public void ParsesVolumesExperimentalOptions()
```

Comment:
Yes. It can be done intentionally in some cases. I don't think we have to validate this user input. Our promise here: we will apply whatever user specifies in option directly to pod/container spec if we have parsed this section